flash/core/model.py

```diff
@@ -163,22 +171,36 @@ def configure_optimizers(self) -> torch.optim.Optimizer:
     def data_pipeline(self) -> DataPipeline:
         # we need to save the pipeline in case this class
         # is loaded from checkpoint and used to predict
-        if not self._data_pipeline:
-            try:
-                # datamodule pipeline takes priority
-                self._data_pipeline = self.trainer.datamodule.data_pipeline
-            except AttributeError:
-                self._data_pipeline = self.default_pipeline()
-        return self._data_pipeline
+        return self._get_pipeline('data')
 
     @data_pipeline.setter
     def data_pipeline(self, data_pipeline: DataPipeline) -> None:
```

Comment:
I find it a bit confusing to have `DataPipeline` and `PostProcessingPipeline` as people might expect a `PreprocessingPipeline`. Worth to iterate on this one.

Reply:
Yes, thought so as well. Basically I named it data_pipeline since it does loading + preprocessing. But fine with changing it as well

Reply:
It also does postprocessing, with `after_uncollate`